src/include/ipc/debug.h

```diff
@@ -0,0 +1,38 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2018 Intel Corporation. All rights reserved.
+ *
+ * Author: Author: Karol Trzcinski <karolx.trzcinski@linux.intel.com>
+ */
+
+#ifndef __IPC_DEBUG_H__
+#define __IPC_DEBUG_H__
+
+#if CONFIG_DEBUG
+
+#include <ipc/header.h>
+#include <stdint.h>
+
+enum sof_ipc_dbg_mem_zone {
+	SOF_IPC_MEM_ZONE_SYS		= 0,	/**< System zone */
+	SOF_IPC_MEM_ZONE_SYS_RUNTIME	= 1,	/**< System-runtime zone */
+	SOF_IPC_MEM_ZONE_RUNTIME	= 2,	/**< Runtime zone */
+	SOF_IPC_MEM_ZONE_BUFFER		= 3,	/**< Buffer zone */
+};
```

Comment:
This now sets in stone the zone classes and names for all SOF implementations. Are we ok to make such a commitment, or should this be more generic? @lgirdwood @mmaka1 @dbaluta ?

Reply:
> This now sets in stone the zone classes and names for all SOF implementations.

It's not so bad, I intentionally create new `SOF_IPC_MEM_ZONE_` instead using `SOF_MEM_ZOEN_`, to make it possible to easily add new zone for particular implementation, just memory usage scan won't work for them without ABI change.

Reply:
@ktrzcinx I guess the use case is that we read a kernel debugFS file (for each zone, which sends the IPC) and then prints the data ?

Reply:
@lgirdwood yes, information printing is one use case, second one is for testing memory leaks, when nothing has change from FW side, client (test) is responsible for data analysis.

Reply:
@ktrzcinx Is there a kernel PR pending here too ? Will need approval from kernel side too.

Reply:
https://github.com/thesofproject/linux/pull/2459